core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -951,15 +951,22 @@ class KafkaApis(val requestChannel: RequestChannel,
               new OffsetFetchResponse(Errors.NONE, authorizedPartitionData.asJava, header.apiVersion)
             }
           } else {
-            val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
-              .partition(authorizeTopicDescribe)
-            val (error, authorizedPartitionData) = coordinator.handleFetchOffsets(offsetFetchRequest.groupId,
-              Some(authorizedPartitions))
+            val (authorizedPartitions, unauthorizedPartitions) =
+              offsetFetchRequest.partitions.asScala.partition(authorizeTopicDescribe)
+            val (error, authorizedPartitionData) =
+              coordinator.handleFetchOffsets(offsetFetchRequest.groupId, Some(authorizedPartitions))
             if (error != Errors.NONE)
               offsetFetchRequest.getErrorResponse(error)
             else {
+              val authorizedAndValidatedPartitionData = authorizedPartitionData.map {
+                case (topicPartition, offsetInfo) =>
+                  if (metadataCache.getPartitionInfo(topicPartition.topic, topicPartition.partition).isEmpty)
```

Comment:
@vahidhashemian I am not strong on the 2nd and 3rd point. Regarding the 1st point, I think the difference is that if partition doesn't exist, the suggested approach will return `OffsetFetchResponse.UNKNOWN_PARTITION`, whereas your approach may return `Errors.GROUP_COORDINATOR_NOT_AVAILABLE`, `Errors.NOT_COORDINATOR_FOR_GROUP` etc. I think the main benefit of swapping the two parts is the readability: it seems more reasonable to return `OffsetFetchResponse.UNKNOWN_PARTITION` if the partition doesn't exist, instead of "if the partition doesn't exist AND the `coordinator.handleFetchOffsets` returns no error."